newdle/api.py

```diff
@@ -428,35 +469,38 @@ def update_participant(args, code, participant_code):
         setattr(participant, key, value)
     if args:
         participant.newdle.update_lastmod()
-    db.session.commit()
-
+    db.session.flush()
     if participant.newdle.notify:
         subject = (
             f'{participant.name} updated their answer for {participant.newdle.title}'
             if is_update
             else f'{participant.name} responded to {participant.newdle.title}'
         )
-        notify_newdle_creator(
-            participant,
-            subject,
-            'replied_email.txt',
-            'replied_email.html',
-            {
-                'update': is_update,
-                'creator': participant.newdle.creator_name,
-                'participant': participant.name,
-                'title': participant.newdle.title,
-                'comment': participant.comment,
-                'answers': [
-                    (timeslot, answer == Availability.ifneedbe)
-                    for timeslot, answer in participant.answers.items()
-                    if answer != Availability.unavailable
-                ],
-                'summary_link': url_for(
-                    'newdle_summary', code=participant.newdle.code, _external=True
-                ),
-            },
-        )
+        try:
+            notify_newdle_creator(
+                participant,
+                subject,
+                'replied_email.txt',
+                'replied_email.html',
+                {
+                    'update': is_update,
+                    'creator': participant.newdle.creator_name,
+                    'participant': participant.name,
+                    'title': participant.newdle.title,
+                    'comment': participant.comment,
+                    'answers': [
+                        (timeslot, answer == Availability.ifneedbe)
+                        for timeslot, answer in participant.answers.items()
+                        if answer != Availability.unavailable
+                    ],
+                    'summary_link': url_for(
+                        'newdle_summary', code=participant.newdle.code, _external=True
+                    ),
+                },
+            )
+        except ConnectionRefusedError:
+            logging.error('Failed notifying the newdle creator', exc_info=True)
```

Comment:
`app.logger.exception(...)`

- we use the flask logger in other places as well
- `.exception()` already includes the exc_info by default